### PR TITLE
Rename Katib NAS RL algorithm to ENAS

### PR DESCRIPTION
**Pull request by andreyvelich**
Part of https://github.com/kubeflow/katib/issues/1211.
Ref PR: https://github.com/kubeflow/katib/pull/1143.

I renamed NASRL to ENAS, fixed some links to API docs and added algorithm settings for ENAS.

/assign @jlewi 
/cc @gaocegege @johnugeorge

### CODE REVIEW

File: content/en/docs/components/hyperparameter-tuning/experiment.md [hunk 1]
```diff
@@ -149,7 +146,7 @@ descriptions on this page:
 * [Bayesian optimization](#bayesian)
 * [HYPERBAND](#hyperband)
 * [Hyperopt TPE](#tpe-search)
-* [NAS based on reinforcement learning](#nas)
+* [Efficient Neural Architecture Search (ENAS)](#enas)
```
Comment (gaocegege) [inline content/en/docs/components/hyperparameter-tuning/experiment.md:149]: I think we are not using ENAS, we do not share parameters. Our approach is part of ENAS.

Comment (andreyvelich) [reply content/en/docs/components/hyperparameter-tuning/experiment.md:149]: @gaocegege Should we name it Neural Architecture Search based on ENAS. 
What do you think?

Comment (gaocegege) [reply content/en/docs/components/hyperparameter-tuning/experiment.md:149]: SGTM. We can say this algorithm is ENAS-based

Comment (andreyvelich) [reply content/en/docs/components/hyperparameter-tuning/experiment.md]: Done.

**Pull request merged**